src/Core/Grid/Data/Factory/CustomerGridDataFactoryDecorator.php

```diff
@@ -91,6 +92,7 @@ public function getData(SearchCriteriaInterface $searchCriteria)
     private function applyModifications(RecordCollectionInterface $customers)
     {
         $modifiedCustomers = [];
+        $idn = new InternationalizedDomainNameConverter();
```

Comment:
Since you use this class in several services, I guess you could also declare it as a service and inject it?